src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -79,16 +74,38 @@ export const getThorTradeQuote: GetThorTradeQuote = async ({ deps, input }) => {
         }),
       )
 
-    const quoteRate = await getTradeRate({
+    const quote = await getQuote({
       sellAsset,
       buyAssetId: buyAsset.assetId,
       sellAmountCryptoBaseUnit,
       receiveAddress,
       deps,
     })
 
-    const rate = await quoteRate.match({
-      ok: rate => Promise.resolve(rate),
+    const slippagePercentage = quote.isOk()
+      ? bn(quote.unwrap().slippage_bps).div(1000)
+      : bn(DEFAULT_SLIPPAGE).times(100)
+
+    const rate = await quote.match({
+      ok: quote => {
+        const THOR_PRECISION = 8
+        const expectedAmountOutThorBaseUnit = quote.expected_amount_out
+        // Add back the outbound fees
+        const expectedAmountPlusFeesCryptoThorBaseUnit = bn(expectedAmountOutThorBaseUnit).plus(
+          quote.fees.outbound,
+        )
+        const sellAmountCryptoPrecision = bn(sellAmountCryptoBaseUnit).div(
+          bn(10).pow(sellAsset.precision),
```

Comment:
we can use `baseUnitToPrecision` if you prefer

Reply:
Addressed in https://github.com/shapeshift/web/pull/4361/commits/14a26c4702a6da0f999cdb4a3987ad25bf7091b2.